Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm

```diff
@@ -172,8 +172,12 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
         }
     }
 
-    if (RefPtr pageClient = this->pageClient())
+    if (RefPtr pageClient = this->pageClient()) {
         pageClient->didCommitLayerTree(layerTreeTransaction);
+#if ENABLE(SCREEN_TIME)
+        pageClient->didChangeScreenTimeWebpageControllerURL();
```

Comment:
This doesn't seem like the right place. You're doing this every single commit, which happens a lot. Don't you want it up next to `m_hasUpdatedRenderingAfterDidCommitLoad = true`, so you only do it for the first remote layer tree commit after didCommitLoad?

Reply:
I believe you are right. Will change now!